coordinator/src/tests/tributary/sync.rs

```diff
@@ -112,21 +114,28 @@ async fn sync_test() {
   assert!(syncer_tributary.read().await.tip().await != syncer_tip);
 
   // Verify it's now participating in consensus
-  // Because only `t` validators are used in a commit, check several commits
-  // This should be biased in favor of the syncer since we're using the syncer's view of the commit
-  for _ in 0 .. 10 {
-    let syncer_tributary = syncer_tributary.read().await;
-    if syncer_tributary
-      .reader()
-      .parsed_commit(&syncer_tributary.tip().await)
-      .unwrap()
-      .validators
-      .iter()
-      .any(|signer| signer == &syncer_key.to_bytes())
-    {
-      return;
-    }
-    sleep(Duration::from_secs(block_time)).await;
+  // Because only `t` validators are used in a commit, take n - t nodes offline
+  // leaving only `t` nodes. Which should force it to participate in the consensus
+  // of next blocks.
+  let n = (spec.n() - spec.t()) as usize;
+  for t in p2p_threads.iter().take(n) {
+    t.abort();
```

Comment:
I'd prefer
```
  let spares = usize::from(spec.n() - spec.t());
  for thread in p2p_threads.iter().take(spares) {
    thread.abort();
```

`as` violates safety, and then you call this `n` when it isn't `n`. It's `n-t`. Then the usage of a variable, `t`, for a thread when `t` already exists is in conflict. But it's not worth blocking the merge for if tests pass.